database_cleanup/models/purge_columns.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# © 2014-2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import _, api, fields, models
+from openerp.exceptions import UserError
+from openerp.osv import fields as legacy_fields
+
+
+class CleanupPurgeLineColumn(models.TransientModel):
+    _inherit = 'cleanup.purge.line'
+    _name = 'cleanup.purge.line.column'
+
+    model_id = fields.Many2one('ir.model', 'Model', required=True,
+                               ondelete='CASCADE')
+    wizard_id = fields.Many2one(
+        'cleanup.purge.wizard.column', 'Purge Wizard', readonly=True)
+
+    @api.multi
+    def purge(self):
+        """
+        Unlink columns upon manual confirmation.
+        """
+        for line in self:
+            if line.purged:
+                continue
+            model_pool = self.env[line.model_id.model]
+            # Check whether the column actually still exists.
+            # Inheritance such as stock.picking.in from stock.picking
+            # can lead to double attempts at removal
+            self.env.cr.execute(
+                'SELECT count(attname) FROM pg_attribute '
+                'WHERE attrelid = '
+                '( SELECT oid FROM pg_class WHERE relname = %s ) '
+                'AND attname = %s',
+                (model_pool._table, line.name))
+            if not self.env.cr.fetchone()[0]:
+                continue
+
+            self.logger.info(
+                'Dropping column %s from table %s',
+                line.name, model_pool._table)
+            self.env.cr.execute(
+                """
+                ALTER TABLE "%s" DROP COLUMN "%s"
+                """ % (model_pool._table, line.name))
+            line.write({'purged': True})
+            self.env.cr.commit()
```

Comment:
Thanks for clarify!
following the rule:
`All cr.commit() calls outside of the server framework from now on must have an explicit comment explaining why they are absolutely necessary, why they are indeed correct, and why they do not break the transactions...`

You can add `# pylint: disable=invalid-commit` too to avoid the check:
database_cleanup/models/purge_columns.py:47: [E8102(invalid-commit), `CleanupPurgeLineColumn.purge] Use of cr.commit() directly - More info https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#never-commit-the-transaction`